src/components/Link/RouterLink.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { ComponentProps } from './Link.types';
+import { Link as ReactRouterLink } from 'react-router-dom';
+
+import classNames from 'classnames';
+import styles from './Link.module.scss';
+
+export interface RouterLinkProps extends ComponentProps {
+  to: string;
+  replace?: boolean;
```

Comment:
nit: We can use the type from LinkProps

```
import type { LinkProps } from 'react-router-dom';

export interface RouterLinkProps extends ComponentProps {
  to: LinkProps['to'];
  replace: LinkProps['replace'];
}
```

This is the LinkProps definition - https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/index.tsx#L341-L349

Reply:
We might need to pass `pathname` and `state` into `to` props.

Reply:
Good call!